softlayer/resource_softlayer_global_ip.go

```diff
@@ -124,7 +125,22 @@ func resourceSoftLayerGlobalIpUpdate(d *schema.ResourceData, meta interface{}) e
 		return fmt.Errorf("Not a valid global ip ID, must be an integer: %s", err)
 	}
 
-	_, err = service.Id(globalIpId).Route(sl.String(d.Get("routes_to").(string)))
+	routes_to := d.Get("routes_to").(string)
+	if strings.Contains(routes_to, ":") && len(routes_to) != 39 {
+		if colonCount := strings.Count(routes_to, ":"); colonCount != 7 {
+			routes_to = strings.Replace(routes_to, "::", "::"+strings.Repeat(":", 7-colonCount), 1)
+		}
+
+		parts := strings.Split(routes_to, ":")
+		for x, s := range parts {
+			parts[x] = fmt.Sprintf("%04s", s)
+		}
+
+		routes_to = strings.Join(parts, ":")
+		d.Set("routes_to", routes_to)
+	}
+
+	_, err = service.Id(globalIpId).Route(sl.String(routes_to))
```

Comment:
@renier 
1. Conversion will not be necessary in `resourceSoftLayerGlobalIpCreate` since `route_to` operation is not initiated there. `resourceSoftLayerGlobalIpCreate` calls `resourceSoftLayerGlobalIpUpdate` in the end wherein the `routes_to` is actually populated.
2. I did not understand. Are you suggesting update to `resourceSoftLayerGlobalIpRead` or to line 140 `d.Set("routes_to", routes_to)`?
3. Incorporated the modified code for decompressing IPv6 and tested it for creation/updation.